test/e2e/version_test.go

```diff
@@ -28,10 +28,13 @@ import (
 func TestVersion(t *testing.T) {
 	t.Parallel()
 
-	r := test.NewKnRunResultCollector(t)
+	it, err := test.NewKnTest()
+	assert.NilError(t, err)
+
+	r := test.NewKnRunResultCollector(t, it)
 	defer r.DumpIfFailed()
 
-	out := test.Kn{}.Run("version")
+	out := it.Kn().Run("version")
```

Comment:
from [logs](https://prow.knative.dev/view/gcs/knative-prow/pr-logs/pull/knative_client/773/pull-knative-client-integration-tests/1245495159261499396#1:build-log.txt%3A694):
```
--- FAIL: TestVersion (1.20s)
    result_collector.go:75: ERROR: unknown flag: --namespace
    result_collector.go:101: === FAIL: =======================[[ERROR]]========================
        🦆 kn version --namespace kne2etests6
        ┃ 
        🔥 unknown flag: --namespace
        🔥 
        =================================================================
```

Here `test.NewKnTest()` would create a new namespace for this test and `Kn().Run()` would check if namespace is non empty and add `--namespace` flag in the command.

We can replace this test as
```
func TestVersion(t *testing.T) {
  t.Parallel()
  r := test.NewKnRunResultCollector(t, nil)
  defer r.DumpIfFailed()
  out := test.RunKn("", []string{"version"})
  r.AssertNoError(out)
  assert.Check(t, util.ContainsAll(out.Stdout, "Version"))
}
```
run
```
➜  client git:(1de3841e) ✗ go test ./test/e2e/ -count=1 -test.v --tags e2e -run TestVersion
=== RUN   TestVersion
=== PAUSE TestVersion
=== CONT  TestVersion
--- PASS: TestVersion (0.04s)
PASS
ok  	knative.dev/client/test/e2e	0.052s
```

Reply:
Slight change to be cleaner but generally should address